sbws/core/scanner.py

```diff
@@ -478,6 +479,14 @@ def main_loop(args, conf, controller, relay_list, circuit_builder, result_dump,
     measured.
 
     """
+    # Variable to count total progress in the last days:
```

Comment:
Since these are only used for the heartbeat message, how about you track these in the hearbeat module?'

And then instead of doing `[consensus_fp_set.add(r) for r in relay_list.relays_fingerprints]` below, you can add a function to the heartbeat module to register these fingerprints with it (and also clean them if/when they need to be cleaned).

Reply:
Hmm, the idea there was to register all the relays for which we keep measurements, not just the last consensus.
Unless we make a singelton too to keep track of this, i thinks this variable needs to be left there.
The "old" relays fingerprints are already cleaned in `RelayList`.
Maybe we should not care about previous consensuses and calculate the progress with respect the last consensus? (which seems would make this simpler).